internal/semver/compare_test.go

```diff
@@ -0,0 +1,64 @@
+package semver
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("IsUpgradable", func() {
+	// testCases is a map of installed version to latest version to expected result
+	testCases := map[string]map[string]bool{
```

Comment:
Since I am doing something similar, I stumbled across [Table Specs](https://onsi.github.io/ginkgo/#table-specs) for such cases – but it's only syntactic sugar and not necessary to change here.